lib/Conversion/VerifToSMT/VerifToSMT.cpp

```diff
@@ -40,11 +45,74 @@ struct VerifAssertOpConversion : OpConversionPattern<verif::AssertOp> {
     Value cond = typeConverter->materializeTargetConversion(
         rewriter, op.getLoc(), smt::BoolType::get(getContext()),
         adaptor.getProperty());
+    Value notCond = rewriter.create<smt::NotOp>(op.getLoc(), cond);
+    rewriter.replaceOpWithNewOp<smt::AssertOp>(op, notCond);
+    return success();
+  }
+};
+
+/// Lower a verif::AssumeOp operation with an i1 operand to a smt::AssertOp
+struct VerifAssumeOpConversion : OpConversionPattern<verif::AssumeOp> {
+  using OpConversionPattern<verif::AssumeOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(verif::AssumeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Value cond = typeConverter->materializeTargetConversion(
+        rewriter, op.getLoc(), smt::BoolType::get(getContext()),
+        adaptor.getProperty());
     rewriter.replaceOpWithNewOp<smt::AssertOp>(op, cond);
     return success();
   }
 };
 
+// Fail to convert unsupported verif ops to avoid silent failure
+struct VerifCoverOpConversion : OpConversionPattern<verif::CoverOp> {
+  using OpConversionPattern<verif::CoverOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(verif::CoverOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    return op.emitError("Conversion of CoverOps to SMT not yet supported");
+  };
+};
+
+struct VerifClockedAssertOpConversion
+    : OpConversionPattern<verif::ClockedAssertOp> {
+  using OpConversionPattern<verif::ClockedAssertOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(verif::ClockedAssertOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    return op.emitError(
+        "Conversion of ClockedAssertOps to SMT not yet supported");
+  };
+};
+
+struct VerifClockedCoverOpConversion
+    : OpConversionPattern<verif::ClockedCoverOp> {
+  using OpConversionPattern<verif::ClockedCoverOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(verif::ClockedCoverOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    return op.emitError(
+        "Conversion of ClockedCoverOps to SMT not yet supported");
+  };
+};
+
+struct VerifClockedAssumeOpConversion
+    : OpConversionPattern<verif::ClockedAssumeOp> {
+  using OpConversionPattern<verif::ClockedAssumeOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(verif::ClockedAssumeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    return op.emitError(
+        "Conversion of ClockedAssumeOps to SMT not yet supported");
+  };
+};
+
```

Comment:
Not totally sure if this is necessary, but my thinking is that without these patterns the errors produced will likely be pretty opaque to someone not familiar with CIRCT/MLIR internals, which maybe isn't ideal for this tool.